Assets/Scripts/Model/Unit.cs

```diff
@@ -72,4 +67,10 @@ void OnTriggerExit (Collider col)
 			this.weapon.removeTarget (col.gameObject.GetComponent<CanReceiveDamage> ());
 		}
 	}
+
+	public bool isDead ()
+	{
+		return this == null;
```

Comment:
It works, here is the answer to your question from answers.unity3d.com:
"It's because all objects that exist in the scene, including GameObjects and all components, have both a managed version (a version in C#) and an unmanaged, native code version that exists inside the engine. When you get a MissingReferenceException, it is because you are attempting to access an object that still exists in managed memory (because you retained a reference to it in C#, so the garbage collector hasn't de-allocated it yet), but which was de-allocated in the unmanaged part of the engine.

This typically occurs when you call Object.Destroy on something to destroy it in unmanaged memory, and then try to access the object's properties afterwards."